cmd/entrypoint/testutil_fake_filtersecrets_test.go

```diff
@@ -0,0 +1,243 @@
+package entrypoint_test
+
+import (
+	"testing"
+
+	"github.com/datawire/ambassador/v2/cmd/entrypoint"
+	"github.com/datawire/ambassador/v2/pkg/snapshot/v1"
+	snapshotTypes "github.com/datawire/ambassador/v2/pkg/snapshot/v1"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+)
+
+// Predicate used to find snapshots that have the specified Secret
+func HasSecret(name string, namespace string) func(snapshot *snapshot.Snapshot) bool {
+	return func(snapshot *snapshot.Snapshot) bool {
+		for _, s := range snapshot.Kubernetes.Secrets {
+			if s.Name == name && s.Namespace == namespace {
+				return true
+			}
+		}
+		return false
+	}
+}
+
+func TestFindFilterNamespacedSecretV3alpha1(t *testing.T) {
+	t.Setenv("EDGE_STACK", "true")
+	f := entrypoint.RunFake(t, entrypoint.FakeConfig{EnvoyConfig: true, DiagdDebug: true}, nil)
+
+	assert.NoError(t, f.UpsertYAML(`
+---
+apiVersion: getambassador.io/v3alpha1
+kind: Filter
+metadata:
+  name: namespaced-secret-filter
+  namespace: foo
+spec:
+  OAuth2:
+    authorizationURL: https://login.example.com/dummy-client/v2
+    clientID: dummy-id
+    clientURL: https://dummy-client-url.com
+    secretName: namespaced-secret
+    secretNamespace: bar
+---
+apiVersion: v1
+data:
+  dummy-data: IGR1bW15LWRhdGE6IGZvbyAK
+kind: Secret
+metadata:
+  name: namespaced-secret
+  namespace: bar
+type: Opaque
+`))
+	f.Flush()
+
+	// After we flush the above config then secrets should be reconciled which should
+	// result in the above secret that is referenced by our filter being added to the snapshot if everything succeeds
+	snap, err := f.GetSnapshot(HasSecret("namespaced-secret", "bar"))
+	require.NoError(t, err)
+	assert.NotNil(t, snap)
+	t.Setenv("EDGE_STACK", "")
+}
+
+func TestFindFilterNamespacedSecretV2(t *testing.T) {
+	t.Setenv("EDGE_STACK", "true")
+	f := entrypoint.RunFake(t, entrypoint.FakeConfig{EnvoyConfig: true, DiagdDebug: true}, nil)
+
+	assert.NoError(t, f.UpsertYAML(`
+---
+apiVersion: getambassador.io/v2
+kind: Filter
+metadata:
+  name: namespaced-secret-filter
+  namespace: foo
+spec:
+  OAuth2:
+    authorizationURL: https://login.example.com/dummy-client/v2
+    clientID: dummy-id
+    clientURL: https://dummy-client-url.com
+    secretName: namespaced-secret
+---
+apiVersion: v1
+data:
+  dummy-data: IGR1bW15LWRhdGE6IGZvbyAK
+kind: Secret
+metadata:
+  name: namespaced-secret
+  namespace: foo
+type: Opaque
+`))
+	f.Flush()
+
+	// After we flush the above config then secrets should be reconciled which should
+	// result in the above secret that is referenced by our filter being added to the snapshot if everything succeeds
+	snap, err := f.GetSnapshot(HasSecret("namespaced-secret", "foo"))
+	require.NoError(t, err)
+	assert.NotNil(t, snap)
+	t.Setenv("EDGE_STACK", "")
+}
+
+func TestFindFilterSecretNamespaceV2(t *testing.T) {
```

Comment:
What do you think about moving these to a `cmd/entrypoint/secrets_test.go`?

The reason being is that these are true UnitTest where is the ones above are closer to integration test since they depend on `diagd` being available on the host. In general, I have seen this in other places in the code base where we have mixed these which I find a little murky and as a developer and surprising.

Same for all the Test below.